lib/src/cli/install/install_command.dart

```diff
@@ -19,81 +19,401 @@
+    final client = HttpClient();
+    var url = 'https://github.com/realm/realm-dart/releases/download/${realmPubspec.version}/$archiveName';
+    if (debug) {
+      url = 'http://localhost:8000/$archiveName';
+    }
+    try {
+      final request = await HttpClient().getUrl(Uri.parse(url));
+      final response = await request.close();
+      if (response.statusCode >= 400) {
+        throw Exception("Error downloading Realm binaries from $url. Error code: ${response.statusCode}");
       }
+      await response.pipe(destinationFile.openWrite());
+    }
+    //TODO: This does not handle download errors.
+     finally {
+      client.close(force: true);
+    }
+
+    print("Extracting Realm binaries to ${destinationDir.absolute.path}");
+    final archive = Archive();
+    await archive.extract(destinationFile, destinationDir);
 
-      final realmDartPackage = packageConfig.packages
-          .where((p) => p.name == 'realm_dart')
-          .firstOrNull;
-      if (realmDartPackage == null) {
-        throw Exception(
-            "realm_dart package not found in dependencies. Add `realm_dart` package to the pubspec.yaml");
+    await saveVersionFile(destinationDir, realmPubspec);
+  }
+
+  Future<void> downloadAndExtractiOSBinaries(String realmPackagePath, Pubspec realmPubspec) async {
+    var destinationDir = Directory(path.join(path.dirname(realmPackagePath), "ios"));
+    if (await skipDownload(destinationDir.absolute.path, realmPubspec.version.toString())) {
+      return;
+    }
+
+    var archiveName = "ios.tar.gz";
+    final destinationFile = File(path.join(destinationDir.absolute.path, archiveName));
+    if (!await destinationDir.exists()) {
+      await destinationDir.create(recursive: true);
+    }
+
+    print("Downloading Realm binaries for $packageName@${realmPubspec.version} to ${destinationFile.absolute.path}");
+    final client = HttpClient();
+    try {
+      var url = 'https://github.com/realm/realm-dart/releases/download/${realmPubspec.version}/$archiveName';
+      if (debug) {
+        url = 'http://localhost:8000/$archiveName';
       }
+      final request = await HttpClient().getUrl(Uri.parse(url));
+      final response = await request.close();
+      if (response.statusCode >= 400) {
+        throw Exception("Error downloading Realm binaries from $url. Error code: ${response.statusCode}");
+      }
+      await response.pipe(destinationFile.openWrite());
+    }
+    //TODO: This does not handle download errors.
+     finally {
+      client.close(force: true);
+    }
+
+    print("Extracting Realm binaries to ${destinationDir.absolute.path}");
+    final archive = Archive();
+    await archive.extract(destinationFile, destinationDir);
+
+    await saveVersionFile(destinationDir, realmPubspec);
+  }
+
+  Future<void> downloadAndExtractMacOSFlutterBinaries(String realmPackagePath, Pubspec realmPubspec) async {
```

Comment:
It appears that a lot of code is repeated between the various `downloadAndExtractX` methods

Reply:
yes. I will create a task to refactor this as discussed